xray/conventions.py

```diff
@@ -826,10 +826,10 @@ def stackable(dim):
         if decode_coords:
             var_attrs = new_vars[k].attrs
             if 'coordinates' in var_attrs:
-                var_coord_names = var_attrs['coordinates'].split()
+                coords = pop_to(var_attrs, new_vars[k].encoding, 'coordinates')
+                var_coord_names = coords.split()
                 if all(k in variables for k in var_coord_names):
                     coord_names.update(var_coord_names)
-                    del var_attrs['coordinates']
```

Comment:
So, one cute thing about the existing code is that it doesn't delete the `coordinates` attribute unless all the coordinates are valid. That's what the `assertDatasetIdentical` test is doing that you changed to `assertDatasetEqual`.